### PR TITLE
feat: rewrite resolvability checker

### PR DESCRIPTION
**Pull request by Aenimus**
## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

**Review by devsergiy — APPROVED** [review]

File: composition/tests/resolvability.test.ts
Comment (StarpTech) [inline composition/tests/resolvability.test.ts:1]: Do you have tests that verify that we abort at certain recursivity?

File: router-tests/cmd/config-validator/extract_schemas.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"slices"
+
+	"gopkg.in/yaml.v3"
+)
+
+type Configuration struct {
```
Comment (StarpTech) [inline router-tests/cmd/config-validator/extract_schemas.go:15]: Please use the auto generated config types.

**Review by StarpTech — CHANGES_REQUESTED** [review]

**Review by StarpTech — APPROVED** [review]
LGTM

File: router-tests/cmd/config-validator/extract_schemas.go (hunk 1 above)
Comment (Aenimus) [reply router-tests/cmd/config-validator/extract_schemas.go:15]: Discussed and addressed,

**Pull request merged**